pyro/infer/inspect.py

```diff
@@ -43,8 +43,20 @@ def is_sample_site(msg):
     return True
 
 
+def site_is_deterministic(msg: dict) -> bool:
+    return msg["type"] == "sample" and msg["infer"].get("_deterministic", False)
+
+
 class TrackProvenance(Messenger):
     def _pyro_post_sample(self, msg):
+        if site_is_deterministic(msg):
```

Comment:
Can you add a boolean flag `self.include_deterministic` to `TrackProvenance` as in #3259 and check it here:
```python
if self.include_deterministic and site_is_deterministic(msg):
```

Reply:
Should it be also extended to TrackProvenance calling (and in its uses in `get_dependencies`, `get_model_relations`, etc)?

Reply:
Yes, as discussed in #3259 we want to be able to switch between the current behavior of `TrackProvenance` that only tracks stochastic dependencies for use in inference and the new behavior in this PR that tracks deterministic and stochastic dependencies for use in visualization.